test/e2e/common/autoscaling_utils.go

```diff
@@ -99,15 +101,15 @@ func GetResourceConsumerImage() string {
 	return resourceConsumerImage
 }
 
-func NewDynamicResourceConsumer(name, kind string, replicas, initCPUTotal, initMemoryTotal, initCustomMetric int, cpuLimit, memLimit int64, f *framework.Framework) *ResourceConsumer {
-	return newResourceConsumer(name, kind, replicas, initCPUTotal, initMemoryTotal, initCustomMetric, dynamicConsumptionTimeInSeconds,
-		dynamicRequestSizeInMillicores, dynamicRequestSizeInMegabytes, dynamicRequestSizeCustomMetric, cpuLimit, memLimit, f)
+func NewDynamicResourceConsumer(name, nsName, kind string, replicas, initCPUTotal, initMemoryTotal, initCustomMetric int, cpuLimit, memLimit int64, clientset clientset.Interface, internalClientset *internalclientset.Clientset) *ResourceConsumer {
```

Comment:
I don't really like the idea of flattening the Framework parameter and passing the individual clientsets instead.  Can you convince me otherwise?

Reply:
The use of framework here is to consume only the cluster clientset, cluster internalclientset and namespace. The resource consumer, the controller rc that controls the resource consumer and communications across using local service discovery, need only these 3 for their correct functioning. The e2e framework, however has hell a lot many things, including loggers, piping/redirection of stuff, their own beforeeach/aftereach (and many more things I did not explore). The fw we create and use in fed e2e tests is the extention of the e2e fw with subtle differences (for example it stores fed clientset, and clientsets for all clusters). The easier way for me to reuse these rc utils to drive the fed test was to create the resource consumer directly into the specific cluster, I want the utilisation driven into and create a matching object (deployment/rs) in federation to control the distribution. I did explore creating a dummy fw with only the cluster clientset, cluster internalclientset and namespace for that particular cluster set, and use that instead (rather then flattening it); but was afraid of some inbuilt suff on that fw (for example beforeeach/aftereach) interfering with the actual fed e2e fw which is used to drive the fed e2e test.
Meanwhile, I think using the dummy fw, as mentioned above might work here, however, this somehow felt the right way to make these utils not really depend on it.

Reply:
OK, fair enough.  I accept your argument.  Given that, the rest of the PR LGTM.